src/subject_name/verify.rs

```diff
@@ -295,3 +300,148 @@ impl<'a> FromDer<'a> for GeneralName<'a> {
 
     const TYPE_ID: DerTypeId = DerTypeId::GeneralName;
 }
+
+#[cfg(feature = "alloc")]
+impl fmt::Debug for GeneralName<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            GeneralName::DnsName(name) => write!(
+                f,
+                "DnsName(\"{}\")",
+                String::from_utf8_lossy(name.as_slice_less_safe())
+            ),
+            GeneralName::DirectoryName => write!(f, "DirectoryName"),
+            GeneralName::IpAddress(ip) => {
+                write!(f, "IpAddress({:?})", IpAddrSlice(ip.as_slice_less_safe()))
+            }
+            GeneralName::UniformResourceIdentifier(uri) => write!(
+                f,
+                "UniformResourceIdentifier(\"{}\")",
+                String::from_utf8_lossy(uri.as_slice_less_safe())
+            ),
+            GeneralName::Unsupported(tag) => write!(f, "Unsupported({})", tag),
+        }
+    }
+}
+
+#[cfg(feature = "alloc")]
+struct IpAddrSlice<'a>(&'a [u8]);
+
+#[cfg(feature = "alloc")]
+impl fmt::Debug for IpAddrSlice<'_> {
```

Comment:
Is this taken from upstream?

Reply:
No, this is Invented Here with the goal of finding a minimal implementation that can still match the one in Python cryptography (which is used for generating the relevant tests).

The [`core` implementation](https://doc.rust-lang.org/src/core/net/ip_addr.rs.html#2006) seems a bit more involved, not sure that's worth it?

Reply:
That makes sense, thanks!